New branch

Rooman255 opened the pull request:
Надеюсь отправилось

Rooman255 committed:
попытка добавить

Rooman255 committed:
Delete Test.java

Liberate520 commented on src/Tree.java:
```diff
@@ -0,0 +1,28 @@
+        return famyliTree.toString();
+    }
+
+    public void findPerson(String name){
```
метод должен возвращать объект, чтоб с ним можно было в дальнейшем работать или куда то передать, а не выводить в консоль

Rooman255 committed:
Delete src directory

Rooman255 committed:
Add files via upload

Rooman255 commented:
попытка дз2

Liberate520 commented on src/Filehandler.java:
```diff
@@ -0,0 +1,20 @@
+import java.io.*;
+
+public class Filehandler {
```
для этого класса требовалось создать интерфейс для дальнейшей интеграции класса в проект. В интерфейсе два метода

Liberate520 commented on src/Filehandler.java:
```diff
@@ -0,0 +1,20 @@
+import java.io.*;
+
+public class Filehandler {
+    FamilyTree familyTree = new FamilyTree();
+    String fileName;
```
private

Liberate520 commented on src/Filehandler.java:
```diff
@@ -0,0 +1,20 @@
+import java.io.*;
+
+public class Filehandler {
+    FamilyTree familyTree = new FamilyTree();
```
Почему вдруг тут новое дерево то? вообще поле лишнее, так как объекты для сохранения передаются через аргументы метода

Rooman255 committed:
Delete FamilyTree.java

Rooman255 committed:
Delete Filehandler.java

Rooman255 committed:
Delete Gender.java

Rooman255 committed:
Delete Main.java

Rooman255 committed:
Delete Person.java

Rooman255 committed:
Add files via upload

DZ4

Rooman255 committed:
Delete Writable.java

Rooman255 committed:
Delete Main.java

Rooman255 committed:
Delete FileHandler.java

Rooman255 committed:
Delete FamilyTree.java

Rooman255 committed:
Delete person directory

Rooman255 committed:
Delete comporator directory

Rooman255 committed:
Add files via upload

DZ4

Rooman255 commented:
DZ4

Rooman255 committed:
Delete src directory

Rooman255 committed:
Add files via upload

DZ5

Liberate520 commented on src/view/Console.java:
```diff
@@ -0,0 +1,127 @@
+import presenter.Presenter;
+
+public class Console implements View {
+    Presenter presenter;
+    Scanner scn = new Scanner(System.in);
+    String key;
+    String nameFather;
+    String nameMather;
+    String name;
+    String newName;
+    int yearBirth;
+    int yearDeath;
```
забываете про инкапсулцию и столько полей абсолютно точно классу не нужно. Скорей всего это могли бы быть локальные переменные какого то метода. В данном классе достаточно презентера и сканера в качестве полей

Rooman255 committed:
Delete src directory

Rooman255 committed:
Переосмысление и финальное ДЗ

Много эксперементов, постоянно все ломал, переделывал и менял, в итоге получилось что то такое

Rooman255 commented:
ломал, смотрел, думал, снова ломал, но вроде что то получилось

Liberate520 commented on src/service/Service.java:
```diff
@@ -0,0 +1,63 @@
+
+public class Service {
+    private int id;
+    private FamilyTree<Person> familyGroup;
```
Я бы сюда добавил объект Interf(название конечно выбрано неудачно)

Liberate520 replied:
Создание объектв методе нарушает 5 принцип

Liberate520 commented on src/tree/FamilyTree.java:
```diff
@@ -0,0 +1,91 @@
+            if (item.getId() == person.getIdMather()) mather = item;
+            if (item.getId() == person.getIdFather()) father = item;
+        }
+        System.out.printf("Исследуемый объект:\n%s\nМать - %s\nОтец - %s", person, mather, father);
```
Работа с консолью должна быть в классе консоли
